src/basic/unit-name.c

```diff
@@ -211,6 +211,26 @@ int unit_name_to_prefix_and_instance(const char *n, char **ret) {
         return 0;
 }
 
+int unit_name_to_suffix(const char *n, char **ret) {
+        const char *e;
+        char *s;
+
+        assert(n);
+        assert(ret);
+
+        if (!unit_name_is_valid(n, UNIT_NAME_ANY))
+                return -EINVAL;
+
+        assert_se(e = strrchr(n, '.'));
+
+        s = strdup(e);
+        if (!s)
+                return -ENOMEM;
```

Comment:
Hmm, wouldn't we be better off returning a static string? Something like
```C
assert_se(s = unit_type_to_string(unit_type_from_string(e)));
return s;
```
This doesn't have the dot, but it can be easily added by the caller.

Reply:
hmm, so, this is actually pretty bad... currently we have the unit "type" which is without the dot, and the unit "suffix" which includes the dot. The "type" is an enum named `UnitType` plus the usual from_string() + to_string() calls, hence I think it's pretty natural that they come without dots. Now, the suffix is otoh is used whenver we split or put together unit names, most prominently in unit_name_build(), where I think it makes sense to include it...

Grmbl... I see if I find a nice solution for this